pkg/meta/base.go

```diff
@@ -740,6 +740,13 @@ func (m *baseMeta) Link(ctx Context, inode, parent Ino, name string, attr *Attr)
 	if name == "" {
 		return syscall.ENOENT
 	}
+	err := m.GetAttr(ctx, inode, attr)
+	if err != 0 {
+		return err
+	}
+	if (attr.Flags&FlagAppend) != 0 || (attr.Flags&FlagImmutable) != 0 {
```

Comment:
Check this inside the transaction to avoid race and the `GetAttr()`

Reply:
done